addons/sms/models/sms_template.py

```diff
@@ -157,3 +157,7 @@ def _get_translated_bodies(self, res_ids):
     def _render_template(self, template_txt, model, res_ids):
         """ Render the jinja template """
         return self.env['mail.template']._render_template(template_txt, model, res_ids)
+
+    def unlink(self):
+        self.mapped('sidebar_action_id').unlink()
```

Comment:
Side note: are we sure rights to unlink actions is granted ?

Reply:
I think that only admins have access to the technical menu and they can delete actions (see https://i.imgur.com/anhYUVQ.png).

Reply:
Access to technical menu is not something we should rely on. I can directly forge the url to access the sms template form view, or add a custom menu, or even call the method in a script / server action, ...

Always try to separate UI from access to code :) .

Reply:
You need to check the ACL for the deletion of template and context action. If a user who has the right to delete a template has the right to delete a context action, it's good. Otherwise, you'll need a sudo

Reply:
Indeed, it appears that some templates can be deleted by managers (who are not admins) so I'll do that.